Document Packit-as-a-Service

phracek opened the pull request:
Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>

Pull Request adds introductory about Packit-as-a-Service and Installation steps for the GitHub application.

Tested locally with hugo.

TomasTomecek commented on content/packit-as-a-service.md:
```diff
@@ -0,0 +1,46 @@
+---
+title: "Packit-as-a-Service"
```
can we name it Documentation?

phracek replied:
It makes sense to me if `packit.dev` is web only about `Packit-as-a-Service`. But If we talk about `Packit` and `Packit-as-a-Service`, then I wanted to differentiate it a bit. That was my proposal.

TomasTomecek replied:
Yes, that's a good point. In that case, packit-as-a-service makes sense.

On the flip side, I'd say that our end-goal is packit-as-a-service where packit tool is really just a detail of the complete solution.

Oki, let's leave it as it is.

TomasTomecek commented on config.toml:
```diff
@@ -28,6 +28,11 @@ paginate = 5
     url = "/source-git/"
-
+  [[menu.main]]
+    name = "Packit-as-a-Service"
```
How about "Documentation" here?

phracek replied:
The same as above.

TomasTomecek commented on config.toml:
```diff
@@ -28,6 +28,11 @@ paginate = 5
+    pre = "<i class='fa fa-user fa-fw'></i>"
+    weight = 4
+    identifier = "packit-as-a-service"
+    url = "/packit-as-a-service/"
```
and docs/ here

phracek replied:
I see if `Documentation` then `docs`. OK.

TomasTomecek commented on content/packit-as-a-service.md:
```diff
@@ -0,0 +1,46 @@
+draft: false
+---
+
+## Introductory description
```
this line is not needed really

TomasTomecek commented on content/packit-as-a-service.md:
```diff
@@ -0,0 +1,46 @@
+Packit Service validates your pull requests by building your software in Fedora OS.
+Once the builds are done, packit lets you know how to install your change
+inside your environment. Get confidence by trying out your changes yourself
+before shipping them to your users.
```
I'm not sure we need a marketing pitch in the intro/docs, how about putting it to "About"?

phracek replied:
Yeah, it could be. I will move it.

phracek replied:
Does it make sense to create in "About" section two subsections like:

~~~
$ About.md
---
title:
---
## Packit-as-a-Service
marketing

## Packit
info about packit
~~~

jpopelka replied:
I wouldn't pollute "About" with this.
I think Tomas just wanted to say that we don't need such marketing tone.
What I'd suggest:
- remove `### Makes you confident` to merge this with the previous section
- remove the `Get confidence` and merge the sentences into `Once the builds are done, packit lets you know how to install the RPMs inside your environment so you can try the changes before merging them.`

phracek replied:
ok, I would prefer to remove it as well completely. Waiting for other opinions.

jpopelka replied:
> Waiting for other opinions.

there's no one else, we are all alone here :-)

phracek replied:
Like @TomasTomecek. He already reacts by emoji.

phracek commented on content/packit-as-a-service.md:
```diff
@@ -0,0 +1,46 @@
+Your pull requests are built in Fedora version of your choice,
+so you're always sure that your software works. When you're ready for a release,
+packit takes care of all the hassle and is able to ship your new upstream release
+into Fedora Rawhide.
```
- Pull Requests are built in Fedora version, or more correctly, code in pull requests is built in Fedora version.
- `packit takes care` is meant to be CLI. `Packit-as-a-Service` is built on the top of packit CLI tool. The code is performed by `packit`

But I would remove `Introductory description` at all. After a time, I don't see any advantage to have Marketing pitch there.

phracek commented on content/packit-as-a-service.md:
```diff
@@ -0,0 +1,46 @@
+
+Using Packit Service is very straightforward: add one config file to your repository,
+followed by an RPM spec file and you're good to go. We also support packit as a CLI tool,
+so you can always try things locally on your own.
```
After reading several times. `packit as a tool` is also useful. `CLI tool` looks weird.

jpopelka replied:
Ok, what I meant was: This doc is about the GitHub App, so why mention the tool/CLI at all?

We can leave it here, but with a little rewording and link to the tool docs.
"Besides the service, Packit is also a tool/CLI [a link to some doc]"

phracek commented on content/packit-as-a-service.md:
```diff
@@ -0,0 +1,46 @@
+1. From the [Packit-service page](https://github.com/apps/packit-as-a-service) page,
+in the left sidebar, click "Install App"
+2. Click "Install" next to the organization or user account containing the correct repository
+3. Install the "Packit-as-a-Service" on all repositories or select repositories
```
I have changed to `or to certain repositories`

TomasTomecek commented:
It's clear that the documentation provided in this PR is not enough: how should the users continue when they add the app to their project?

jpopelka commented on content/packit-as-a-service.md:
```diff
@@ -0,0 +1,40 @@
+draft: false
+---
+
+Packit-as-a-Service is a GitHub Application (https://developer.github.com/apps/about-apps/).
```
```suggestion
Packit-as-a-Service is a GitHub App (https://developer.github.com/apps/about-apps/).
```

jpopelka commented on content/packit-as-a-service.md:
```diff
@@ -0,0 +1,40 @@
+
+### Easy to use
+
+Using Packit Service is very straightforward: add one config file to your repository,
```
a link do docs please

phracek commented:
> It's clear that the documentation provided in this PR is not enough: how should the users continue when they add the app to their project?
> 
> Edit: but I'd say, let's address that in a next PR and ideally talk about that in person tomorrow (Wed).

Thanks for the comment. I forgot to initiate a meeting what kind of information we would like to have there. You are right.

Once application is installed and PR is created user see status from COPR build and after reviewing the PR can be merged. Of course, we can add there information that SPEC file is needed in upstream repository and packit configuration file with link to documentation . 

What other information about Packit Service should be there? Please point it out and I will added.
Thanks

phracek commented:
I have added three subchapters. Please review my content. I have addressed all @jpopelka issues.
Also, picture how packit-as-a-service  PR looks like is added.

jpopelka commented on content/packit-as-a-service.md:
```diff
@@ -40,3 +40,24 @@ find "Packit-as-a-Service" and select it
+
+### Add SPEC file
+
+Packit Service needs RPM spec file in your upstream repository. This RPM spec file is used for building
```
Could you link https://github.com/packit-service/packit/issues/312 here?

TomasTomecek commented on content/packit-as-a-service.md:
```diff
@@ -0,0 +1,67 @@
+
+Using Packit Service is very straightforward: add one config file to your repository
+([see .packit.yaml format](https://github.com/packit-service/packit/blob/master/docs/configuration.md)),
+followed by an RPM spec file and you're good to go. Besides the service, packit is also as a tool
```
```suggestion
followed by an RPM spec file and you're good to go. Besides the service, packit is also a tool
```

TomasTomecek commented on content/packit-as-a-service.md:
```diff
@@ -0,0 +1,67 @@
+2. Click "Install" next to the organization or user account containing the correct repository
+3. Install the "Packit-as-a-Service" on all repositories or to certain repositories
+
+### Install from GitHub Marketplace
```
since we won't be on marketplace in following days, I'd comment this section out

TomasTomecek commented on content/packit-as-a-service.md:
```diff
@@ -0,0 +1,67 @@
+### Add SPEC file
+
+Packit Service needs RPM spec file in your upstream repository.
+This RPM spec file is used for building project in COPR repositories.
```
```suggestion
This RPM spec file is used to build your project.
```

TomasTomecek commented on content/packit-as-a-service.md:
```diff
@@ -0,0 +1,67 @@
+Packit Service needs RPM spec file in your upstream repository.
+This RPM spec file is used for building project in COPR repositories.
+
+If this makes you barf, you can use a “source git” branch of your upstream repository.
```
WHAT?!

phracek replied:
I have added there from one of our document. Will remove it and RFE remains.

TomasTomecek commented on content/packit-as-a-service.md:
```diff
@@ -0,0 +1,67 @@
+
+### How to check Packit Service works in your project?
+
+Nowadays, Packit Service supports only validation of your pull requests.
```
```suggestion
Nowadays, Packit Service is able only to build your changes and provide them as RPMs.
```
validation implies testing

TomasTomecek commented:
https://github.com/packit-service/packit/pull/386
